### ISSUE TITLE
Failure trying to ensure the CFME user existence

### ISSUE THREAD

**Issue by simon3z** [openshift/openshift-ansible#4555]
#### Description

When re-running the `playbooks/byo/openshift-cfme/config.yml` playbook the task `Ensure the CFME user exists` is stuck for some minutes and then fails with:

```
TASK [openshift_cfme : Ensure the CFME user exists] ****************************
fatal: [ocp-c09-master01.10.35.48.7.nip.io]: FAILED! => {
    "changed": false, 
    "failed": true
}

MSG:

{u'returncode': 1, u'cmd': u'/usr/bin/oc replace -f /tmp/cfme-J0cwKm --force -n default', u'results': {}, u'stderr': u'Error from server (AlreadyExists): object is being deleted: users.user.openshift.io "cfme" already exists\n', u'stdout': u'user "cfme" deleted\n'}
```

Note the `/tmp/cfme-J0cwKm` file is missing so it's impossible to inspect it.

cc @sdodson 

##### Version

```
ansible 2.2.2.0
  config file = /home/simon/src/openshift-ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

```
openshift-ansible-3.6.123-1
```

##### Steps To Reproduce
1. Run `roles/openshift_cfme/tasks/main.yml` twice

##### Expected Results
The playbook should complete.

##### Observed Results
The playbook is stuck on a task for several minutes and fails.

**Comment by tbielawa** [comment]
I ran into a similar issue testing this on 3.6. I couldn't find any other reproducing cases so I just kept it in the back of my mind. I'm glad someone else has independently confirmed this observation (I am still sorry that you had to run into this problem, though). I am not sure if I need to file an issue in upstream OCP or with our openshift-ansible [`oc_user`](https://github.com/openshift/openshift-ansible/blob/master/roles/lib_openshift/library/oc_user.py) module.

For now please note the warning in this section of the CFME readme:
* https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_cfme#important-notes-1

> Most importantly follow this one rule: don't re-run the entrypoint playbook multiple times in a row
> without cleaning up after previous runs if some of the CFME steps have ran. This is a known flake.
> Cleanup instructions are provided at the bottom of this README.


To work around it you will need to preform the [cleanup steps](https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_cfme#manual-cleanup) in the README.

----

After you run the user delete command you should run `oc get users` again. 

```
$ oc delete user cfme
$ oc get users | grep cfme
cfme     <long-uid-string-here>
```

**If the user is still there**, then you'll need to use the manual work-around which is to override the default username assigned to the project.

```ini
# INVENTORY

[OSEv3:vars]
openshift_cfme_user=cfme2
```

----

## In short

1. Sorry about the error, we'll look at that
1. Take note of the warning about running the playbook twice, this is a noted limitation in this tech preview release
1. Run the manual cleanup steps referenced above
1. Double check that the user was actually deleted after you run `oc delete`
1. Override the user name in your inventory if the user will not go away

----

Please let me know what happens so we can track this as tech debt in the next cycle.

**Comment by tbielawa** [comment]
I just realized you said

```
Steps To Reproduce

Run roles/openshift_cfme/tasks/main.yml twice
```

Please ensure you are running the playbook from the [proper entrypoint playbook](https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_cfme#getting-started)

```
$ ansible-playbook -v -i <INVENTORY_FILE> playbooks/byo/openshift-cfme/config.yml
```

**Comment by simon3z** [comment]
cc @gtanzillo @bazulay @fbladilo

**Comment by simon3z** [comment]
@tbielawa I cannot remove the user with `oc delete user cfme` ...do you know why?

**Comment by tbielawa** [comment]
@simon3z I **think** this is an OCP bug. I ran into similar issues. I have no idea what causes it right now. I might try reproducing it using the `oc create user` command manually to get extra verification.

**Comment by tbielawa** [comment]
This is fixed up in the new OCP 3.7 release. Please re-open if there is more you need to discuss.
